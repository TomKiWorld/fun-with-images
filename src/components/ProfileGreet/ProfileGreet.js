import React from 'react';

const ProfileGreet = ({ entries }) => {
  if (entries > 0) {
    return (
        <div>
          <p>Thank you for using the app {entries} times so far!!</p> 
          <h3>The following images were submitted by you:</h3>
          <p>Note: Some images might have been submitted several times</p>
        </div>) 
  } else {
    return <h3>You haven't submitted any images yet..</h3>;
  }
}

export default ProfileGreet;
