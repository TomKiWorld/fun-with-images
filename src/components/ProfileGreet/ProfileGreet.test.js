import { shallow } from 'enzyme';
import React from 'react';
import ProfileGreet from './ProfileGreet';

describe('ProfileGreet Component', () => {
  const wrapper = shallow(<ProfileGreet entries={5} />);
  it('Expect to render ProfileGreet Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to render ProfileGreet Component with entries', () => {
    const title = wrapper.find('h3').at(0);
    expect(title.text()).toBe('The following images were submitted by you:');
  });
  it('Expect to render ProfileGreet Component without entries', () => {
    const wrapper = shallow(<ProfileGreet entries={0} />);
    const title = wrapper.find('h3').at(0);
    expect(title.text()).toBe('You haven\'t submitted any images yet..');
  });
});
