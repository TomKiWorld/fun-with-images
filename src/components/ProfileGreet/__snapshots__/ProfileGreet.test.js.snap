// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileGreet Component Expect to render ProfileGreet Component 1`] = `ShallowWrapper {}`;
